# Assessing
Use the space below to explore `winequality-red.csv` and `winequality-white.csv` to answer the quiz questions below.

Assessing Data
Using Pandas, explore winequality-red.csv and winequality-white.csv in the Jupyter notebook below to answer quiz questions below the notebook about these characteristics of the datasets:

number of samples in each dataset
number of columns in each dataset
features with missing values
duplicate rows in the white wine dataset
number of unique values for quality in each dataset
mean density of the red wine dataset
This data was originally taken from here:
https://archive.ics.uci.edu/ml/datasets/Wine+Quality

In [1]:
import pandas as pd
# maptplotlib inline

df_red = pd.read_csv('winequality-red.csv', sep = ';')
df_white = pd.read_csv('winequality-white.csv',sep = ';')

In [2]:
df_red.info()
df_white.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1599 entries, 0 to 1598
Data columns (total 12 columns):
fixed acidity           1599 non-null float64
volatile acidity        1599 non-null float64
citric acid             1599 non-null float64
residual sugar          1599 non-null float64
chlorides               1599 non-null float64
free sulfur dioxide     1599 non-null float64
total sulfur dioxide    1599 non-null float64
density                 1599 non-null float64
pH                      1599 non-null float64
sulphates               1599 non-null float64
alcohol                 1599 non-null float64
quality                 1599 non-null int64
dtypes: float64(11), int64(1)
memory usage: 150.0 KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4898 entries, 0 to 4897
Data columns (total 12 columns):
fixed acidity           4898 non-null float64
volatile acidity        4898 non-null float64
citric acid             4898 non-null float64
residual sugar          4898 non-null float64

In [3]:
df_red.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5  
3      9.8        6  
4      9.4        5

In [4]:
#number of duplicate rows
sum(df_red.duplicated())

240

In [5]:
#number of duplicate rows
sum(df_white.duplicated())

937

In [7]:
#number of unique values of quality
print("Unique values of quality for red wine are {}".format(len(df_red.quality.unique())))
print("unique values of quality for white wine are {}".format(len(df_white.quality.unique())))

Unique values of quality for red wine are 6
unique values of quality for white wine are 7


In [8]:
df_red['density'].mean()

0.9967466791744833

# Questions and Answers to Quiz

Answers derived by code calculations provided above

- How many samples of red wine are there? 1599
- How many samples of white wine are there? 4898
- How many columns are in each dataset? 12
- Which features have missing values? None of these features have missing values
- How many duplicate rows are in the white wine dataset? 937
- Are duplicate rows in these datasets significant/ need to be dropped? Not necessarily
- How many unique values of quality are in the red wine dataset? 6
- How many unique values of quality are in the white wine dataset? 7
- What is the mean density in the red wine dataset? 0.996747


